include/glow/Backends/TraceEvents.h

```diff
@@ -161,6 +166,22 @@ class TraceContext {
   void
   logTraceEvent(llvm::StringRef name, llvm::StringRef type, uint64_t timestamp,
                 std::map<std::string, std::string> additionalAttributes = {});
+
+  /// Sets the human readable \p name for thread \tid.
+  void setThreadName(int tid, llvm::StringRef name);
+  /// \returns the list of human readable thread names.
+  std::map<int, std::string> &getThreadNames() { return threadNames_; }
+
+  /// Dumps all TraceEvents in json format to the given \filename, optionally
+  /// with a provided \p processName.
+  void dump(llvm::StringRef filename, std::string processName = "");
```

Comment:
const ref here for processName.

Reply:
\p filename

Reply:
Can't have a const ref default param.

Reply:
https://godbolt.org/z/NepmJQ

Reply:
huh. ok, guess I'm wrong let me take another look.